apps/cart/views.py

```diff
@@ -43,13 +54,29 @@ def add_to_cart(self, request):
         response_data = CartItemSerializer(cart_item)
         return Response(response_data.data, status=status.HTTP_201_CREATED)
 
-    @action(detail=False, methods=['delete'], url_path='remove/(?P<item_id>[^/.]+)')
+    @action(detail=False, methods=["get"], url_path="cartitem/(?P<item_id>[^/.]+)")
+    def retrieve_cartitem(self, request, item_id=None):
+        try:
+            cart_item = CartItem.objects.get(id=item_id)
+        except CartItem.DoesNotExist:
+            return Response(
+                {"detail": "Item not found"}, status=status.HTTP_404_NOT_FOUND
+            )
+
+        response_data = CartItemSerializer(cart_item)
+        return Response(response_data.data, status=status.HTTP_202_ACCEPTED)
+
+    @action(detail=False, methods=["delete"], url_path="remove/(?P<item_id>[^/.]+)")
     def remove_from_cart(self, request, item_id=None):
         try:
             cart_item = CartItem.objects.get(id=item_id)
 
         except CartItem.DoesNotExist:
```

Comment:
Whenever you use `objects.get` 2 possible errors may occur DoesNotExist and MultipleObjectsReturned.. you need to handle both